opencti-platform/opencti-graphql/src/domain/container.js

```diff
@@ -52,27 +48,37 @@ export const numberOfContainersForObject = (context, user, args) => {
 };
 
 export const objects = async (context, user, containerId, args) => {
-  const key = buildRefRelationKey(RELATION_OBJECT, '*');
   const types = args.types ? args.types : ['Stix-Core-Object', 'stix-relationship'];
+  const relationArgs = { fromId: containerId, toTypes: types, baseData: true };
+  const objectRelationsPromise = listAllRelations(context, user, RELATION_OBJECT, relationArgs);
+  const key = buildRefRelationKey(RELATION_OBJECT, '*');
   const filters = [{ key, values: [containerId], operator: 'wildcard' }, ...(args.filters || [])];
-  const data = args.all ? await paginateAllThings(context, user, types, R.assoc('filters', filters, args))
-    : await listThings(context, user, types, R.assoc('filters', filters, args));
+  const queryFilters = { ...args, filters, connectionFormat: false };
+  const elementsPromise = args.all ? listAllThings(context, user, types, queryFilters) : listThings(context, user, types, queryFilters);
```

Comment:
Why are we retrieve elements from another query rather than retrieve them from the relations ?
We can have const toIds = relations.map((rel) => rel.toId) and resolve toIds, no ?

Reply:
Its a general problem of the current model. In order to support case of pagination/sorting/filtering we use the denormalized id of elements stored in rel_. Unfortunalty we miss the fact that storing also the direction was needed and so when we list through that part we have sometime to much elements.